corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -295,60 +348,81 @@ hqDefine("geospatial/js/geospatial_map", [
-                cases = contextData.cases;
-                map.addCaseMarkersToMap();
-
-                if (contextData.invalid_geo_cases_report_link) {
-                    var missingCasesLink = contextData.invalid_geo_cases_report_link;
-                    var missingCasesLinkTag = "<a href=" + missingCasesLink + ">" + gettext("View here") + "</a>";
-                    var message = gettext("There are case(s) missing geolocation data.");
-
-                    alert_user.alert_user(message + " " + missingCasesLinkTag, "warning");
-
-                    var $bannerAlert = $("#message-alerts");
-                    if ($bannerAlert.children().length > 1) {
-                        // Remove the initial banner, since it contains the old link
-                        $bannerAlert.children()[0].remove();
-                    }
-                }
-            }
-
             if ($mapControlDiv.length) {
-                ko.cleanNode($mapControlDiv);
+                ko.cleanNode($mapControlDiv[0]);
                 $mapControlDiv.koApplyBindings(mapControlsModelInstance);
             }
 
+            var $saveDrawnArea = $("#btnSaveDrawnArea");
             $saveDrawnArea.click(function(e) {
                 if (map) {
                     saveGeoJson(map.getMapboxDrawInstance(), mapControlsModelInstance);
                 }
             });
+
+            var $exportDrawnArea = $("#btnExportDrawnArea");
             $exportDrawnArea.click(function(e) {
                 if (map) {
                     mapControlsModelInstance.exportGeoJson()
                 }
             });
+        };
+
+        var missingGPSModel = function(cases) {
+            this.casesWithoutGPS = ko.observable(cases);
+        };
+        var missingGPSModelInstance = new missingGPSModel([]);
+
+        $(document).ajaxComplete(function (event, xhr, settings) {
+            const isAfterReportLoad = settings.url.includes('geospatial/async/case_management_map/');
+            // This indicates clicking Apply button or initial page load
+            if (isAfterReportLoad) {
+                map = loadMapBox();
+                initMapControls();
+                // Hide controls until data is displayed
+                showMapControls(false);
+                return;
+            }
+
+            // This indicates that report data is fetched either after apply or after pagination
+            const isAfterDataLoad = settings.url.includes('geospatial/json/case_management_map/');
+            if (!isAfterDataLoad) {
+                return;
+            }
+            isDataNeverFetched = true;
+            showMapControls(true);
+            // Hide the datatable rows but not the pagination bar
+            $('.dataTables_scroll').hide();
+
+            if (xhr.responseJSON.aaData.length && map) {
+                map.removeMarkers();
+                var casesWithGPS = xhr.responseJSON.aaData.filter(function(item) {
+                    return item[1] != null;
```

Comment:
What's your thoughts on having a mapping somewhere to give the expected field indecies names? This way, we can have something this that might read better?
```Javascript
return item[expectedFields.coordinates] != null;
```

Just a suggestion, but it's up to you if you want to do it or not :smiley:

Reply:
I wish the same, but, unfortunately the data returned by the view must be in this tuple format (instead of the mapping), because the underlying datatables expects that format. Since the UI uses datatables for the pagination, the mapping has to work with it, so there's no way out without refactoring the datatables which is complicated and which would be out of scope for this PR.

Reply:
Ah ok, thanks for clarifying. All good!